activemq-console/src/main/java/org/apache/activemq/console/command/store/StoreBackup.java

```diff
@@ -178,7 +194,19 @@ public boolean recoverMessage(Message message) throws IOException {
                         return true;
                     }
                 };
-                if(offset != null) {
+                if(startMsgId != null || endMsgId != null) {
+                    System.out.println("Backing up from startMsgId: " + startMsgId + " to endMsgId: " + endMsgId);
+                    queue.recoverNextMessages(startMsgId, endMsgId, (count != null ? count : Integer.MAX_VALUE), queueRecoveryListener, true);
+                } else if(indexesList != null) {
+                    System.out.println("Backing up using indexes count: " + indexesList.size());
+                    for(int idx : indexesList) {
+                        if(idx < 0) {
```

Comment:
Why would there be a negative index? I would think this should be an exception and not just a continue